UI/PopulationPanel.cpp

```diff
@@ -70,7 +70,30 @@ void PopulationPanel::CompleteConstruction() {
     std::vector<std::pair<MeterType, MeterType>> meters;
 
     for (auto& meter_stat : m_meter_stats) {
-        meter_stat.second->InstallEventFilter(shared_from_this());
+        meter_stat.second->RightClickedSignal.connect([this, &meter_stat](const GG::Pt& pt) {
+            MeterType meter_type = meter_stat.first;
+
+            std::string meter_string = boost::lexical_cast<std::string>(meter_type);
+
+            auto popup = GG::Wnd::Create<CUIPopupMenu>(pt.x, pt.y);
+            auto pc = this->GetPopCenter();
+            if (meter_type == METER_POPULATION && pc) {
+                std::string species_name = pc->SpeciesName();
+                if (!species_name.empty()) {
+                    auto zoom_species_action = [&species_name]() { ClientUI::GetClientUI()->ZoomToSpecies(species_name); };
```

Comment:
Seemingly the problem is here, when capturing `species_name` by reference, not by value. It's a local, so is probably out of scope and/or reset to an empty string by a later loop iteration by the time the lambda gets executed.

Removing the `&` from the capture fixes it for me.

Reply:
should probably do all the captures in these loops by value...?

Reply:
Updated PR according to suggestion.